### DISCUSSION TITLE
Deno.run hangs when a child process returns more than 8200 characters and the stdout == piped.

### DISCUSSION THREAD

**Discussion by CarlosOnline** [denoland/deno#12638] [Q&A]
`Deno.run` **hangs** when a child process returns **more than 8200 characters** and the **stdout == piped**.

Repro source code is in repository: https://github.com/CarlosOnline/deno-bugs.git

# Repro:
- git clone `https://github.com/CarlosOnline/deno-bugs.git`
- cd deno-bugs
- `deno.exe run -A repro-run-hang.ts`

**Deno will hang between 8100 and 8200 characters.**

# Test code:
```
"a".repeat(parseInt(Deno.args[0]))
```

# Repro code:
```
/**
 * Execute Deno.run with file that generates specified length of characters.
 * @param length Length of output to generate.
 */
async function Run(length: number) {
  console.log(`Running with length ${length}`);

  const p = Deno.run({
    cmd: [
      "deno",
      "run",
      "--allow-read",
      ".\\write-output.ts",
      length.toString(),
    ],
    stdout: "piped",
    stderr: "piped",
  });

  const { code } = await p.status();
  const rawOutput = await p.output();
  console.log(`Exit code: ${code} length: ${rawOutput.length}`);
}

/**
 * Loop through different lengths until hangs or succeeds.
 */
for (let idx = 100; idx < 10000; idx += 100) {
  await Run(idx);
}

console.log("Success");
```

**Comment by piscisaureus** [comment]
You have to read from stdout to avoid this.
Your program is waiting for the child process to exit first (await p.status).

**Comment by CarlosOnline** [comment]
I don't understand, the Deno example shows to wait for the status.
Plus the code works for output less than 8200 characters long.

https://deno.land/manual@v1.12.2/examples/subprocess

Should I skip `const { code } = await p.status();` the line instead?  

```
const { code } = await p.status();

// Reading the outputs closes their pipes
const rawOutput = await p.output();
const rawError = await p.stderrOutput();
```

**Comment by lucacasonato** [comment]
@CarlosOnline Stdout only has a limited maximum size that will buffered in the kernel. If your subprocess continually writes to the stdout pipe, it will eventually fill up. This will prevent the sub-process from exiting (because it is now blocked on writing its data to stdout, but the kernel won't let it). To prevent the pipe filling up, you need to read some bytes from your parent process. One way you can do this, is by running `p.output()` before `p.status`, or by running them in parallel (using `Promise.all` or similar).
